### PR TITLE
Integrate ESLint into Design System

### PR DESCRIPTION
**Pull request by sdalonzo**
- Add ESLint
- Add ESLint pre-commit hook
- Fix low hanging fruit
- Add ignores for other issues to be addressed later

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    node: true,
+  },
+  extends: [
+    'eslint:recommended',
+    'plugin:react/recommended',
+    'plugin:jest/style',
+    'pcln-accessibility',
+    'prettier',
+    // 'plugin:@typescript-eslint/eslint-recommended',
+    // 'plugin:@typescript-eslint/recommended',
```
Comment (craigpalermo) [inline .eslintrc.js:14]: ESLint requires a bit of extra configuration to work with TypeScript, so it'll probably make sense to keep these commented out until we support `.tsx?` files outside of the `types` project.

File: docs/IconField.md [hunk 2]
```diff
@@ -45,7 +45,7 @@ It will parse children based on the `isField` and `isIcon` static properties.
 By default the design system `Input`, `Select`, and `Icon` components will work with the `IconField` component, but to allow other children to render, add an appropriate static property.
 
 ```jsx
-import React from 'react'
+import * as React from 'react'
```
Comment (craigpalermo) [inline docs/IconField.md:48]: Do we need the changed import syntax since we're not turning on TS plugins?

Comment (sdalonzo) [reply docs/IconField.md:48]: maybe not? I thought I remembered reading that this was the new best practice..is that only for TS?

File: .eslintrc.js (hunk 1 above)
Comment (sdalonzo) [reply .eslintrc.js:14]: sure, makes sense

Comment (sdalonzo) [reply .eslintrc.js:14]: these came as part of `eslint --init` when prompted for TS support. i figured it was easier to add them now and comment it out, then to have to piece it back together later.

**Pull request merged**